gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDatasetTest.java

```diff
@@ -216,6 +223,40 @@ public void testGetFilesAtPath() throws IOException {
       Assert.assertTrue(candidateFiles.contains(PathUtils.getPathWithoutSchemeAndAuthority(fileStatus.getPath()).toString()));
     }
 
+    // test ds of daily/yyyy-MM-dd-HH-mm-ss
+    datePattern = "yyyy-MM-dd-HH-mm-ss";
+    formatter = DateTimeFormat.forPattern(datePattern);
+    endDate = LocalDateTime.now(DateTimeZone.forID(TimeAwareRecursiveCopyableDataset.DEFAULT_DATE_PATTERN_TIMEZONE));
+
+    candidateFiles = new HashSet<>();
+    for (int i = 0; i < MAX_NUM_DAILY_DIRS; i++) {
```

Comment:
do we not want to create # directories = min(lookback, max_num_daily_dirs)? there could be more directories created than we want if max_num_daily_dirs is > lookback time right.

Reply:
You're right but its just for generating folders for testing which gets cleaned up afterwards. And we would want to test scenarios where there are multiple folders but the lookback covers only a subset of them.